src/osdatahub/NamesAPI/names_api.py

```diff
@@ -0,0 +1,160 @@
+from collections.abc import Iterable
+from typing import Union
+
+import requests
+from geojson import FeatureCollection
+from typeguard import typechecked
+
+from osdatahub.NamesAPI.local_types import validate_local_type, get_local_type
+from osdatahub.extent import Extent
+from osdatahub.grow_list import GrowList
+from osdatahub.utils import addresses_to_geojson
+
+
+class NamesAPI:
+    """Main class for querying the OS Names API (https://osdatahub.os.uk/docs/names/overview)
+
+    Args:
+        key (str): A valid OS API Key. Get a free key here - https://osdatahub.os.uk/
+
+    Example::
+
+        from osdatahub import FeaturesAPI, Extent
+        from os import environ
+
+        key = environ.get("OS_API_KEY")
+        names = NamesAPI(key)
+        results = features.find("Buckingham Palace", limit=5)
+    """
+    __ENDPOINT = r"https://api.os.uk/search/names/v1/"
+    HEADERS = {"method": "POST",
+               "headers": "{'Content-Type': 'application/json'}"}
+
+    def __init__(self, key: str):
+        self.key = key
+
+    def __endpoint(self, api_name: str) -> str:
+        return self.__ENDPOINT + api_name + f"?key={self.key}"
+
+    @typechecked
+    def find(self, text: str, limit: int = 100,
+             bounds: Extent = None, bbox_filter: Extent = None,
```

Comment:
it was indeed to distinguish from bbox object. It would be confusing to have the argument bbox be type Extent. Do you think it could be useful to change the name of the BBox object instead?

Reply:
There's something to be said for it actually expecting a BBox object, since that's what it wants (a generic extent still works, but the code takes its BBox extent anyway!).

Reply:
The bbox_filter method must be in British National Grid CRS, so it's good to be able to check for that. BBox doesn't allow for CRS's, so I'm not sure we want to go for the less feature-rich BBox object